test/ts/unit/collateralizer/runners/return_collateral.ts

```diff
@@ -0,0 +1,232 @@
+                await mockTermsContract.mockTermEndTimestamp.sendTransactionAsync(
+                    scenario.agreementId,
+                    new BigNumber(scenario.termEndTimestamp),
+                );
+
+                if (typeof scenario.before !== "undefined") {
+                    await scenario.before(collateralizer, mockTermsContract);
+                }
+
+                ABIDecoder.addABI(collateralizer.abi);
+            });
+
+            after(() => {
+                ABIDecoder.removeABI(collateralizer.abi);
+            });
+
+            if (scenario.succeeds) {
+                it("should not throw", async () => {
+                    txHash = await collateralizer.returnCollateral.sendTransactionAsync(
+                        scenario.agreementId,
+                        { from: scenario.from(COLLATERALIZER, NON_COLLATERALIZER) },
+                    );
+
+                    expect(txHash.length).to.equal(66);
+                });
+
+                it("should erase record of current collateralization", async () => {
+                    await expect(
+                        collateralizer.agreementToCollateralizer.callAsync(
+                            scenario.agreementId,
+                        ),
+                    ).to.eventually.equal(NULL_ADDRESS);
+                });
+
+                it("should emit log that collateral has been returned", async () => {
+                    const receipt = await web3.eth.getTransactionReceipt(txHash);
+                    const [CollateralSeizedLog] = compact(ABIDecoder.decodeLogs(receipt.logs));
+
+                    expect(CollateralSeizedLog).to.deep.equal(
+                        CollateralReturned(
+                            collateralizer.address,
+                            scenario.agreementId,
+                            COLLATERALIZER,
+                            mockCollateralToken.address,
+                            scenario.collateralAmount,
+                        ),
+                    );
+                });
+
+                it("should transfer collateral from collateralizer contract to owner", async () => {
+                    await expect(
+                        mockCollateralToken.wasTransferCalledWith.callAsync(
+                            COLLATERALIZER,
+                            scenario.collateralAmount,
+                        ),
+                    ).to.eventually.be.true;
+                });
+            } else {
+                it("should throw", async () => {
+                    await expect(
+                        collateralizer.returnCollateral.sendTransactionAsync(
+                            scenario.agreementId,
+                            { from: scenario.from(COLLATERALIZER, NON_COLLATERALIZER) },
+                        ),
+                    ).to.eventually.be.rejectedWith(REVERT_ERROR);
+                });
+            }
+        });
+    }
+
+    private packTermsContractParameters(
```

Comment:
This seems like it should be a part of the `terms_contract_parameters.ts` factory